### PR TITLE
Centralize publishing configuration

### PR DESCRIPTION
**Pull request by Enkelian**

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    `kotlin-dsl`
+}
+
+repositories {
+    mavenCentral()
+    google()
+}
+
+dependencies {
+    implementation("com.android.tools.build:gradle:7.1.0")
```
Comment (mateuszrzeszutek) [inline buildSrc/build.gradle.kts:11]:
```suggestion
    implementation("com.android.tools.build:gradle:7.1.1")
```

File: splunk-otel-android-volley/build.gradle.kts [hunk 2]
```diff
@@ -1,10 +1,8 @@
-import java.net.URI
 import java.time.Duration
 
 plugins {
     id("com.android.library")
-    id("maven-publish")
-    id("signing")
+    id("splunk.publish-conventions")
```
Comment (mateuszrzeszutek) [inline splunk-otel-android-volley/build.gradle.kts:5]: Nit: WDYT about calling it `splunk.android-library-conventions`? This plugin does a bit more than just setting up artifact publishing

File: buildSrc/src/main/kotlin/splunk.publish-conventions.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        }
+    }
+    publications {
+        register<MavenPublication>("maven") {
+            groupId = "com.splunk"
+            artifactId = base.archivesName.get()
+
+            pom {
+                name.set("Splunk Otel Android")
+                description.set("A library for instrumenting Android applications for Splunk RUM")
```
Comment (mateuszrzeszutek) [inline buildSrc/src/main/kotlin/splunk.publish-conventions.gradle.kts:29]: Shouldn't we set these in the libraries' `build.gradle.kts` files? I think the volley lib could have a bit different description

**Review by mateuszrzeszutek — COMMENTED** [review]
Left couple of minor comments, overall it LGTM 👍

File: buildSrc/src/main/kotlin/splunk.publish-conventions.gradle.kts (hunk 3 above)
Comment (Enkelian) [reply buildSrc/src/main/kotlin/splunk.publish-conventions.gradle.kts]: Done, although I wonder if I approached it correctly with setting the name through an `extra`. WDYT?

Comment (mateuszrzeszutek) [reply buildSrc/src/main/kotlin/splunk.publish-conventions.gradle.kts]: I think it's fine; the upstream repo handles a similar use case in a similar way: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/conventions/src/main/kotlin/otel.javaagent-instrumentation.gradle.kts#L8

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**